Atmel/Winc15x0/GHIElectronics.TinyCLR.Drivers.Atmel.Winc15x0/Winc15x0Interface.cs

```diff
@@ -0,0 +1,166 @@
+using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Runtime.CompilerServices;
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+using GHIElectronics.TinyCLR.Devices.Network;
+using GHIElectronics.TinyCLR.Devices.Network.Provider;
+
+namespace GHIElectronics.TinyCLR.Drivers.Atmel.Winc15x0 {
+    public class Winc15x0Interface : INetworkControllerProvider {
+
+        public enum PowerSave {
+            PowerSave_None = 0,
+            PowerSave_Automatic = 1,
+            PowerSave_H_Automatic = 2,
+            PowerSave_Deep_Automatic = 3,
+            PowerSave_Manual = 4,
+        }
+
+        private readonly NetworkController networkController;
+
+        public event NetworkLinkConnectedChangedEventHandler NetworkLinkConnectedChanged {
+            add {
+                this.networkController.NetworkLinkConnectedChanged += value;
+            }
+
+            remove {
+                this.networkController.NetworkLinkConnectedChanged -= value;
+            }
+        }
+        public event NetworkAddressChangedEventHandler NetworkAddressChanged {
+            add {
+                this.networkController.NetworkAddressChanged += value;
+            }
+
+            remove {
+                this.networkController.NetworkAddressChanged -= value;
+            }
+        }
+
+        public NetworkInterfaceType InterfaceType => this.networkController.InterfaceType;
+        public NetworkCommunicationInterface CommunicationInterface => this.networkController.CommunicationInterface;
+
+        public Winc15x0Interface() => this.networkController = NetworkController.FromName("GHIElectronics.TinyCLR.NativeApis.ATWINC15xx.NetworkController");
+
+        ~Winc15x0Interface() => this.Dispose();
```

Comment:
If we're not doing anything except calling SuppressFinalize then a finalizer and dispose are worthless